### PR TITLE
GSoC pages should be more up-to-date and clear

### PR DESCRIPTION
**Pull request by solomax**

### CODE REVIEW

File: source/gsoc.md
Comment (turbaszek) [inline source/gsoc.md:31]:
```suggestion
to the projects you are interested in. 
Or simply use this link https://issues.apache.org/jira/issues/?jql=labels+%3D+gsoc2021 .</span>
```
Keeping this in separate line may improve rendering and readability. WDYT?

Comment (solomax) [reply source/gsoc.md]: Should look better now :)
Thanks!

File: source/guide-to-being-a-mentor.md
Comment (tristaZero) [inline source/guide-to-being-a-mentor.md:15]: 1. `if your project does not use JIRA you can [use the Comdev Issue Tracker For GSoC Tasks](use-the-comdev-issue-tracker-for-gsoc-tasks.html`
2. `if your project doesn't use JIRA contact dev@community.apache.org`

Hi, it looks like these statements above will confuse users since the condition is the same, but there are two solutions for me to choose? 
If we want those mentors whose community dose not use JIRA to use `Comdev Issue Tracker`, maybe it is unnecessary to tell them to ask this question in dev@ml? (I guess if so, we will receive many emails with the same questions.  😃 )
Also, a similar statement appears in `gsoc.md`.

File: source/gsoc.md
Comment (tristaZero) [inline source/gsoc.md:30]: BTW, do you think it will be kind to provide any link for `committers/GSoCLinkId.txt` on line 145? Sorry... actually, I do not know where the file is.

**Review by tristaZero — COMMENTED** [review]
Hi @solomax Thanks for your update, It looks much clear. Also I gave some comments, If there is any questions, please ping me. : )

File: source/guide-to-being-a-mentor.md
Comment (solomax) [reply source/guide-to-being-a-mentor.md]: Thanks @tristaZero,

could you please check the new version? is it more clear?

File: source/gsoc.md
Comment (solomax) [reply source/gsoc.md]: I guess this is in SVN, not sure why is it here

**Review by turbaszek — APPROVED** [review]

File: source/gsoc.md
Comment (tristaZero) [inline source/gsoc.md:148]: Hi @solomax One similar expression here. Please give it attention. Except for that, these guidelines make sense to me. Nice job. 😉

Comment (solomax) [reply source/gsoc.md:148]: I leave it copy/pasted due I'm unsure how to `reuse` the code
I'll merge this version if you don't have any better idea :)))

**Review by tristaZero — APPROVED** [review]

**Pull request merged**
